### DISCUSSION TITLE
How do you get PStates working for overclocked CPU?

### DISCUSSION THREAD

**Discussion by mrdc** [AppleBreak1/EP45-UD3P-Customac#5] [General]
Hi,

I see that you have Intel Core 2 Quad Q9550 @4.0GHz, which was heavily overclocked and you don't use SSDTs with changed P-States/C-States. And these states are not in DSDT you use in Opencore.
Do you have fully working Speed Step: when cores can have different multipliers and intermediate states.
I have CPU PM in Ventura, but Speed Step is quite limited: only Lowest and Highest multipliers, no multipliers in between.

Thanks!

**Comment by AppleBreak1** [comment]
Hello there again. To answer your questions,


> Do you have fully working Speed Step
<p align="center">
<img width="1023" alt="Q9550" src="https://github.com/AppleBreak1/EP45-UD3P-Customac/assets/97265013/d0437cdf-721b-4054-a816-7053aba28023">
</p>

> I see that you have Intel Core 2 Quad Q9550 @4.0GHz, which was heavily overclocked and you don't use SSDTs with changed P-States/C-States.

<p align="center">
<img width="596" alt="2" src="https://github.com/AppleBreak1/EP45-UD3P-Customac/assets/97265013/714a1587-a659-415f-a9e3-8ea969b3fb60">
</p>

  Note: Irrelevant to your question but I have it backed down to 3.2GHz given that the machine has aged and I didn't like the fan noise it was making under load, but yeah it is more than 40% boost in frequency and it makes huge difference when gaming.

> these states are not in DSDT you use in Opencore.

I didn't include them as they can vary from CPU to CPU depending on the model; hence, I linked the guide for it. Even on my SSDT EFI build(All-In-One), I have them disabled, but the SSDTs are there as an example. 

Assuming that you have the exact same CPU model (Mine has E0 Stepping) you may still use those speedstep SSDTs with the patched DSDT or include the scopes in your patched DSDT.  Nonetheless, attached is the slimmed down version of my own disassembled DSDT specific to my need. I thought I had posted a separate DSDT with speedstep scope included as an example but maybe it was removed. 

[DSDT.dsl.zip](https://github.com/AppleBreak1/EP45-UD3P-Customac/files/13125724/DSDT.dsl.zip)

> I have CPU PM in Ventura, but Speed Step is quite limited: only Lowest and Highest multipliers, no multipliers in between.

From my testing, this depended on the SMBIOS used.  More specifically, only the Mac Models with Wolfdale/Penryn/Hapertown will perform with all the defined P states for my Q9550. Outside of these models or when you use DummyPowerManagement, frequency will only hit max or min.

**Comment by mrdc** [comment]
Thanks! I have Lynnfield i5-750 in one of my workstations. Will make SSDT with custom PStates for it.
BTW Have you used some tool for generating PStates? Like VoodooPState kext?

**Comment by mrdc** [comment]
> Note: Irrelevant to your question but I have it backed down to 3.2GHz given that the machine has aged and I didn't like the fan noise it was making under load, but yeah it is more than 40% boost in frequency and it makes huge difference when gaming.

How did you make PSChanger to show your real, overclocked, frequency? For me it shows stock freq: 
![image](https://github.com/AppleBreak1/EP45-UD3P-Customac/assets/16876990/aacd277b-8633-4308-943e-7ba82473c0a3)
